src/bp-integrations/learndash/learndash/Hooks.php

```diff
@@ -45,6 +45,32 @@ public function init() {
 		add_action( 'update_post_meta', array( $this, 'checkLearndashCourseUpdateMeta' ), 10, 4 );
 		add_action( 'added_post_meta', array( $this, 'checkLearndashCourseUpdateMeta' ), 10, 4 );
 		add_action( 'deleted_post_meta', array( $this, 'checkLearndashCourseDeleteMeta' ), 10, 4 );
+		add_filter( 'bp_ps_field_can_filter', array( $this, 'checkLearndashCourseFilter' ), 10, 2 );
+	}
+
+	/**
+	 * Skip the leardash course field from member search when course price type or course mode is 'open'.
+	 * 
+	 * @since BuddyBoss 1.5.9
+	 * 
+	 * @param bool  $has_access Default activity for leardash course search field.
+	 * @param mixed $field      Field attributes for learndash course.
+	 * 
+	 * @uses learndash_get_setting() Get learndash individual course settings.
+	 * 
+	 * @return bool
+	 */
+	public function checkLearndashCourseFilter( $has_access, $field ) {
+		if ( 'field_learndash_courses' === $field->code ) {
+			$course_settings = learndash_get_setting( $field->value );
+			
+			// Skip the leardash course field from member search when course price type or course mode is 'open'.
+			if ( 'open' == $course_settings['course_price_type'] ) {
```

Comment:
```suggestion
			if ( 'open' === $course_settings['course_price_type'] ) {
```
Check for a strict comparision